src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.sps.data.Meal;
+
+@WebServlet("/meal/*")
+public class MealServlet extends HttpServlet {
+
+    private ArrayList<Meal> dishes;
+
+    @Override
+    public void init() {
+        dishes = new ArrayList<>(Arrays.asList());
```

Comment:
Given that we're going to fetch items by id, I'd suggest storing items in a [HashMap](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html) rather than a list. It's got two advantages:
1. The relation between the key and the value is clear. Whenever you add an item to a hashmap, you have to specify what the key is gonna be.
2. Performance is superior if you need to look an item up by an id. For a hashmap, the complexity is O(1). For a list, it's O(n) (you can see it in your implementation below - to find a single item, you have to iterate the whole list, hence O(n))

Reply:
Then do we still need `id` field in class `Meal`?

Reply:
That's a good question.
I'd say "no", to avoid having two sources of the same data.